File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {
    private final int CARS_PER_SPEED = 221;

    public double productionRatePerHour(int speed) {
        return CARS_PER_SPEED * speed * getSuccessRate(speed);
    }

    public int workingItemsPerMinute(int speed) {
        return (int) (productionRatePerHour(speed) / 60);
    }

    private double getSuccessRate(int speed) {
        if (speed >= 1 && speed <= 4)
            return 1.0;
        if (speed >= 5 && speed <= 8)
            return 0.9;
        if (speed == 9)
            return 0.8;
        return 0.77;
    }
}

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
using System;

static class AssemblyLine
{
    private const int ProductionPerSpeed = 221;
    public static double SuccessRate(int speed)
    {
        if(speed < 1){
            return 0;
        }
        if(speed < 5){
            return 1.0;
        }
        if(speed < 9){
            return 0.90;
        }
        if(speed < 10){
            return 0.80;
        }
        return 0.77;
    }
    
    public static double ProductionRatePerHour(int speed)
    {
        return ProductionPerSpeed * speed * SuccessRate(speed);
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return (int)ProductionRatePerHour(speed) / 60;
    }
}
